google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -619,20 +621,40 @@ def _on_response(self, response):
 
         with self._pause_resume_lock:
             for received_message in response.received_messages:
-                message = google.cloud.pubsub_v1.subscriber.message.Message(
-                    received_message.message,
-                    received_message.ack_id,
-                    received_message.delivery_attempt,
-                    self._scheduler.queue,
-                )
-                self._messages_on_hold.put(message)
-                self._on_hold_bytes += message.size
-                req = requests.LeaseRequest(
-                    ack_id=message.ack_id,
-                    byte_size=message.size,
-                    ordering_key=message.ordering_key,
-                )
-                self.leaser.add([req])
+                if (
+                    "googclient_OpenTelemetrySpanContext"
+                    in received_message.message.attributes.keys()
```

Comment:
Dicts support direct containment check, i.e. `"foo" in my_dict`, we do not need to obtain the keys first with the `keys()` method.